### ISSUE TITLE
Fork: Use Jackson instead of org.json

### ISSUE THREAD

**Issue by DavidBiesack** [skyscreamer/JSONassert#49]
We have a toolchain based on Jackson; I'd like to use JSONassert with Jackson JSON objects. I've forked the skyscreamer/JSONassert repo and replaced org.json with [FasterXML/jackson-core](https://github.com/FasterXML/jackson-core) and [FasterXML/jackson-databind](https://github.com/FasterXML/jackson-databind). 

The biggest hurdle is the unit tests; Jackson is more strict and does not allow unquoted object names and unquoted values; but requires double quotes. This requires updating all the unit tests to use quoted names/values such as 

```
{"a":[{"background":"white","id":1,"type":"row"} ...
```

instead of

```
{a:[{background:white,id:1,type:row}
```

(While Jackson has [Feature.ALLOW_UNQUOTED_FIELD_NAMES](https://fasterxml.github.io/jackson-core/javadoc/2.4/com/fasterxml/jackson/core/JsonParser.Feature.html#ALLOW_UNQUOTED_FIELD_NAMES), it does not have a Feature to allow unquoted _values_.)

My real question is: is that fork reasonable to push back to skyscreamer/JSONassert, perhaps in a different branch, or would it be better to push the fork elsewhere as a separate project (because current users of skyscreamer/JSONassert may depend on using unquoted strings)? If I do publish a different fork on GitHub, would you prefer I change the package names as well? I would certainly provide attribution to skyscreamer/JSONassert as the base of the fork, and release it under the same Apache 2.0 license.

thanks.

**Comment by ebourg** [comment]
@DavidBiesack Any chance you could push your Jackson based JSONassert on GitHub? I'd be interested in using a version that doesn't depend on the non-free json.org library.

**Comment by DavidBiesack** [comment]
@ebourg I'd like the JSONAssert maintainer to answer my question as to how/where the fork should go, here as a PR or branch, or elsewhere. Have not heard anything in 14 months, tho...

**Comment by jgoldhammer** [comment]
+1 for changing it...

**Comment by puntogil** [comment]
hi
any new aboutremovel o non free json.org?
as reported in https://github.com/skyscreamer/JSONassert/issues/44 all linux distribution can't use Your library (Fedora in my case)
thanks in advance

**Comment by DavidBiesack** [comment]
See PR #67 **Change the implementation for org.json to one with a more open license** 
This uses an API compatible clean-room implementation of the org.json classes, so the impact is much lower than switching to Jackson. While this still makes it more difficult to use JSONassert with other tools that use Jackson, it is possible and the license would not be a problem.

If that PR is merged, I'll close this issue.

**Comment by carterpage** [comment]
Sorry for the delay. I've merged #67. I don't plan to support Jackson.
